spec/models/grda_warehouse/client_file_spec.rb

```diff
@@ -153,37 +153,44 @@
     let!(:user) { create :user }
     let!(:can_manage_client_files) { create :role, can_manage_client_files: true }
     let!(:can_manage_window_client_files) { create :role, can_manage_window_client_files: true }
+    let!(:can_see_own_file_uploads) { create :role, can_see_own_file_uploads: true }
+    let!(:can_generate_homeless_verification_pdfs) { create :role, can_generate_homeless_verification_pdfs: true }
     let(:config) { create :config }
     let!(:verified_homeless_tag) { create :available_file_tag, name: 'Homeless Verification', verified_homeless_history: true }
-    let!(:file) { create :client_file }
-    let!(:own_file) { create :client_file, user: user }
+    # verified homeless history created by another user
+    let!(:history_file) { create :client_file }
+    # verified homeless history created by own user
+    let!(:own_history_file) { create :client_file, user: user, client: history_file.client }
+    # other type of file created by another user, for another client
+    let!(:other_file) { create :client_file }
```

Comment:
I added another file to the spec (NOT a homeless history) to make sure that the `sql = sql.or(has_full_housing_release.and(is_not_verified_homeless_history))` pieces are working correctly.